### PR TITLE
Update to risc0 pre-0.18

### PR DESCRIPTION
**Pull request by mothran**
Updated the new SNARK Receipt format and completed e2e testing with bonsai and local testnet.

### CODE REVIEW

File: relay/src/lib.rs [hunk 1]
```diff
@@ -15,22 +15,14 @@
...
-    Execution {
-        journal: Vec<u8>,
-    },
-    Bonsai {
-        journal: Vec<u8>,
-        receipt_metadata: ReceiptMetadata,
-        snark_proof: SnarkProof,
-    },
+    Execution { journal: Vec<u8> },
+    Bonsai { snark_proof: SnarkReceipt },
```
Comment (nategraf) [inline relay/src/lib.rs:25]:
```suggestion
    Bonsai { snark_receipt: SnarkReceipt },
```

**Review by nategraf — COMMENTED** [review]
Looks good

File: relay/src/lib.rs (hunk 1 above)
Comment (mothran) [reply relay/src/lib.rs:25]: Thanks! Fixed in: 7ba6cb4ab87ab9ef8f758a5eee457508859e191f

Comment (mothran) [reply relay/src/lib.rs]: I mean: ce7ce12348455a8d3a0512d60432c06eebb39a11

**Review by nategraf — APPROVED** [review]

**Pull request merged**